contentcuration/contentcuration/models.py

```diff
@@ -806,14 +831,17 @@ def save(self, *args, **kwargs):
         if self.cloned_source is None:
             self.cloned_source = self
 
-        if self.original_channel_id is None:
-            # TODO: This SIGNIFICANTLY slows down the creation flow
-            channel = (self.parent and self.parent.get_channel()) or self.get_channel() # Check parent first otherwise new content won't have root
-            self.original_channel_id = channel.id if channel else None
-        if self.source_channel_id is None:
-            # TODO: This SIGNIFICANTLY slows down the creation flow
-            channel = (self.parent and self.parent.get_channel()) or self.get_channel() # Check parent first otherwise new content won't have root
-            self.source_channel_id = channel.id if channel else None
+        # Getting the channel is an expensive call, so warn about it so that we can reduce the number of cases in which
+        # we need to do this.
+        if not channel_id:
```

Comment:
Can also add `and (not self.original_channel_id or not self.source_channel_id)` for faster save